jdr/WFLY-13539-galleon_layer_jdr_support.adoc

```diff
@@ -0,0 +1,131 @@
+
+[source,xml]
+----
+<plugin>
+    <groupId>org.wildfly.plugins</groupId>
+    <artifactId>wildfly-jar-maven-plugin</artifactId>
+    <configuration>
+        <feature-pack-location>wildfly@maven(org.jboss.universe:community-universe)#${version.wildfly}</feature-pack-location>
+        <layers>
+            <layer>jdr</layer>
+            <layer>core-tools</layer>
+        </layers>
+    </configuration>
+    <executions>
+        <execution>
+            <goals>
+                <goal>package</goal>
+            </goals>
+        </execution>
+    </executions>
+</plugin>
+----
+
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13539[WFLY-13539 Add a Galleon layer for JBoss Diagnostic Reporting]
+* https://issues.redhat.com/browse/EAP7-1528[EAP7-1528 Add a Galleon layer for JBoss Diagnostic Reporting]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFGP-177[WFGP-177 Create a Galleon package for the JBoss Diagnostic Reporting bin scripts]
+
+=== Dev Contacts
+
+* mailto:brian.stansberry@redhat.com[Brian Stansberry]
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:fburzigo@redhat.com[Fabio Burzigotti]
+* mailto:jkasik@redhat.com[Jan Kasik]
+
+=== Testing By
+
+* [X] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+https://github.com/wildfly/wildfly[WildFly], https://github.com/wildfly/galleon-plugins[WildFly Galleon Plugins]
+
+=== Other Interested Projects
+
+N/A
+
+== Requirements
+
+=== Hard Requirements
+
+* The galleon layer must provide the provisioning of the JDR subsystem.
+* The provisioning tool will be responsible for providing the bash scripts to generate the diagnosis information - i.e. `jdr.{sh,bat,ps1}` - and make those available in the following filesystem location: `$WFLY_HOME/bin/`.
+* The galleon layer will depend optionally on `secure-management` layer. This layer will allow the users to configure a secure management interface to connect to the server via `jdr.sh`. If a user wants to connect via CLI, they have to add `core-tools` layer to their provision configuration.
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+
+== Test Plan
+
+The test coverage of the Galleon layer added by this proposal is divided in two main groups:
+
+. Testing the Galleon layer provisioning. This testing is done by https://github.com/wildfly/wildfly/blob/master/testsuite/layers/src/test/java/org/jboss/as/test/layers/LayersTestCase.java[LayersTestCase]. The testsuite will be modified to add a new server provisioned with this layer in isolation and with this layer combined with all the layers. For each kind of provisioning, this test does the following:
+.. Verifies the provisioned modules are the expected ones.
+.. Verifies the provisioned server starts successfully.
```

Comment:
See my last reply/questions in the conversation section. Basically I asked whether the tested combinations will be consist of 1+1 (i.e. the tested layer plus one of the remaining layers). This is how I understand it.
In such case, I think that we should add this detail to the section here.

E.g.: s/`new server provisioned with this layer in isolation and with this layer combined with all the layers`/`new server provisioned with this layer in isolation and with this layer combined with each one of the remaining available layers`/

WDYT?

Reply:
Is this point cleared up with https://github.com/wildfly/wildfly-proposals/pull/311#discussion_r461726969?

There is no `1+1` pair combinations, is one combination that include the current layer under test + all the remaining available layers.

Reply:
Ok, right, then the text is correct. Thanks again @yersan!